src_c/mouse.c

```diff
@@ -298,6 +300,20 @@ mouse_set_cursor(PyObject *self, PyObject *args)
     SDL_SetCursor(cursor);
     SDL_FreeCursor(lastcursor);
 
+    //To make sure that the mask data sticks around, it has to have its ref count increased
+    //Conversely, the old data stored in cursor_data (if it is there) doesn't need to be around anymore
+    Py_XDECREF(cursor_data.xormask);
```

Comment:
Can you please describe how the lifetime of the objects is supposed to work?

- Especially with xormask/andmask.
- How are they allocated, and how are they freed?

Reply:
Yes.

The xormask and andmask are received as pointers to python objects, and I decided to store them that way rather than the parsed data in c itself. (So I could just return the objects as received). The problem was, since python didn't know I was storing these objects, it garbage collected the data away, since if it sometimes saw 0 references to them. 

So the new xormask and andmask have their reference counts get manually increased by 1.
Similarly, the data that is being replaced, the old xormask and andmask pyobject pointers, aren't needed by pygame anymore so they can have their reference count decreased by one. If something else still needs them, they will still stick around since their reference count will reflect other references to them.

Having to explain this makes me realize that surfaces have the same bug, since they are stored similarly.

The issue is when things are written "inline" since then there are no other references to them after the function executes.

    pygame.mouse.set_cursor((10,10), pygame.Surface((20,20)))
    a = pygame.mouse.get_cursor()
    print(a) # fails

Reply:
Sounds like there is a reference count issue, I haven't had to look into it in detail yet.
Putting this code into a test would be a nice addition.